server/src/internalClusterTest/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplainIT.java

```diff
@@ -1239,7 +1242,10 @@ private ClusterAllocationExplanation runExplain(boolean primary, String nodeId,
             XContentBuilder builder = JsonXContent.contentBuilder();
             builder.prettyPrint();
             builder.humanReadable(true);
-            logger.debug("--> explain json output: \n{}", Strings.toString(explanation.toXContent(builder, ToXContent.EMPTY_PARAMS)));
+            logger.debug(
+                "--> explain json output: \n{}",
+                Strings.toString(ChunkedToXContent.wrapAsToXContent(explanation).toXContent(builder, ToXContent.EMPTY_PARAMS))
+            );
```

Comment:
I think we don't need to create our own `XContentBuilder` here any more, this whole block could just be as follows:

```
        if (logger.isDebugEnabled()) {
            logger.debug(
                "--> explain json output: \n{}",
                Strings.toString(explanation, true, true)
            );
        }
```